transport/src/test/java/io/netty/channel/socket/Tun4PacketTest.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.socket;
+
+import com.google.common.primitives.Ints;
+import io.netty.buffer.ByteBuf;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static io.netty.buffer.Unpooled.wrappedBuffer;
+import static io.netty.channel.socket.Tun4Packet.INET4_FLAGS_DONT_FRAGMENT_MASK;
+import static io.netty.channel.socket.Tun4Packet.INET4_FLAGS_MORE_FRAGMENTS_MASK;
+import static io.netty.channel.socket.Tun4Packet.INET4_TYPE_OF_SERVICE_DELAY_MASK;
+import static io.netty.channel.socket.Tun4Packet.INET4_TYPE_OF_SERVICE_PRECEDENCE_ROUTINE;
+import static io.netty.channel.socket.Tun4Packet.INET4_TYPE_OF_SERVICE_RELIBILITY_MASK;
+import static io.netty.channel.socket.Tun4Packet.INET4_TYPE_OF_SERVICE_THROUGHPUT_MASK;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class Tun4PacketTest {
```

Comment:
The other test classes have a `public` modifier (e.g., `ChannelInitializerTest`).